set.mm

```diff
@@ -204730,6 +204740,118 @@ with the same (relevant) properties is also a group.  (Contributed by
       JQVGVJVK $.
   $}
 
+  ${
+    $d .+ i j k t x y $.  $d .+^ a b c d f i j k x y $.  $d F a d f i j k x y $.
+    $d G a d f i j k t u x y $.  $d H a b c d f i $.  $d X i j k t u x y $.
+    $d Y a b c d f i j k x y $.  $d a b c i j k x y ph $.
+    ghmgrp.f $e |- ( ( ph /\ x e. X /\ y e. X )
+      -> ( F ` ( x .+ y ) ) = ( ( F ` x ) .+^ ( F ` y ) ) ) $.
+    ${
+      $d A x y $.  $d B y $.
+      mhmlem.a $e |- ( ph -> A e. X ) $.
+      mhmlem.b $e |- ( ph -> B e. X ) $.
+      $( Lemma for ~ mhmmnd and ~ ghmgrp .  (Contributed by Paul Chapman,
+         25-Apr-2008.) (Revised by Mario Carneiro, 12-May-2014.) (Revised by
+         Thierry Arnoux, 25-Jan-2020.) $)
+      mhmlem $p |- ( ph -> ( F ` ( A .+ B ) ) = ( ( F ` A ) .+^ ( F ` B ) ) )
+        $=
+        ( wcel co cfv wceq w3a wi cv eleq1 id 3anbi2d oveq1 fveq2d fveq2 oveq1d
+        eqeq12d imbi12d 3anbi3d oveq2 oveq2d vtocl2g syl2anc imp syl13anc ) AAA
+        DIMZEIMZDEFNZHOZDHOZEHOZGNZPZAUAZVDKLAAUPUQQZVCAUPUQVEVCRZKLABSZIMZCSZI
+        MZQZVGVIFNZHOZVGHOZVIHOZGNZPZRAUPVJQZDVIFNZHOZUTVOGNZPZRVFBCDEIIVGDPZVK
+        VRVQWBWCVHUPAVJVGDITUBWCVMVTVPWAWCVLVSHVGDVIFUCUDWCVNUTVOGVGDHUEUFUGUHV
+        IEPZVRVEWBVCWDVJUQAUPVIEITUIWDVTUSWAVBWDVSURHVIEDFUJUDWDVOVAUTGVIEHUEUK
+        UGUHJULUMUNUO $.
+    $}
+
+    ghmgrp.x $e |- X = ( Base ` G ) $.
+    ghmgrp.y $e |- Y = ( Base ` H ) $.
+    ghmgrp.p $e |- .+ = ( +g ` G ) $.
+    ghmgrp.q $e |- .+^ = ( +g ` H ) $.
+    ghmgrp.1 $e |- ( ph -> F : X -onto-> Y ) $.
+    ${
+      mhmmnd.3 $e |- ( ph -> G e. Mnd ) $.
+      ${
+        $d .0. a d i x y $.
+        mhmid.0 $e |- .0. = ( 0g ` G ) $.
+        $( The neutral element of the image of a monoid by an homomorphism.
+           (Contributed by Thierry Arnoux, 25-Jan-2020.) $)
```

Comment:
Do you mean: "A surjective monoid morphism preserves units."

Reply:
I'll keep your wording, but write "A surjective monoid morphism preserves _identity elements_.", since this is the way it is named elsewhere.